domain/schema/model/sql/0022-port-ranges.sql

```diff
@@ -0,0 +1,36 @@
+CREATE TABLE protocol (
+    id INT PRIMARY KEY,
+    protocol TEXT NOT NULL
+);
+
+INSERT INTO protocol VALUES
+(0, 'icmp'),
+(1, 'tcp'),
+(2, 'udp');
+
+CREATE TABLE port_range (
+    protocol_id INT NOT NULL,
+    from_port INT,
+    to_port INT,
+    unit_endpoint_uuid TEXT NOT NULL,
+    -- We disallow overlapping port ranges on a unit's endpoint.
+    -- This should be validated in the service layer. This means
+    -- we only need to include one end of the port range to
+    -- ensure uniqueness
```

Comment:
Even though I know you're technically correct, I have a gut feeling this will bite us.

Can we add both, it's not like it's going to cost us much. It seems lazy to omit it.

Reply:
Personally, I don't mind either way

What do you think @manadart ?

Reply:
If we add both, then it's a XOR situation making it unique. I would rather have the constraint as is, but the comment should indicate not that "we only need", but that we can't ensure a non-overlapping port range and this is as far as we are leaning into the impedance mismatch at the schema level.

Reply:
The solution is to have a field that encompasses both the start and end, which would get your uniqueness, a natural keep for the port-ranges. This does duplicate information, but it prevents the XOR situation and it gives you a unique primary key.

I'm fine with picking one, as I know this could potentially bite us in the future. Although, I don't care enough to warrant the change, just that there is a solution.